src/GameplaySystem.cpp

```diff
@@ -401,6 +402,26 @@ void GameplaySystem::handle_object_transform_event(const Event& e) {
             )
         );
     }
+
+    glm::vec3 it_pos = glm::vec3(
+                           object_positions_[current_it_id_].x,
+                           object_positions_[current_it_id_].y,
+                           object_positions_[current_it_id_].z
+                       );
+
+    glm::vec3 vec_to = glm::normalize(glm::vec3(it_pos - glm::vec3(x, y, z)));
```

Comment:
We can reduce the glm operations here. object_positions_[current_it_id] returns a glm::vec3, so no need to recreate one. Also, you can use the "-" operator between two glm::vec3 and get a glm::vec3 still.

You probably want something like this:
```
glm::vec3 vec_to = glm::normalize(object_positions[current_it_id_] - object_positions_[object_id]);
```

Of course, you can give them names by defining local variables if you prefer but I think the above snippet conveys the intention clearly.